lib/driver.js

```diff
@@ -100,6 +100,7 @@ class XCUITestDriver extends BaseDriver {
       let { device, udid, realDevice } = await this.determineDevice();
       log.info(`Determining device to run tests on: udid: '${udid}', real device: ${realDevice}`);
       this.device = device;
```

Comment:
We have to pass this down to WebDriverAgent.js Thats why I added it like `this.opts.device = device`. Do you want to keep both `this.device` `this.opts.device`?